src/runtime/pkg/oci/utils.go

```diff
@@ -919,6 +927,57 @@ func addHypervisorNetworkOverrides(ocispec specs.Spec, sbConfig *vc.SandboxConfi
 	})
 }
 
+func addHypervisorInitdataOverrides(ocispec specs.Spec, sbConfig *vc.SandboxConfig) error {
+	if value, ok := ocispec.Annotations[vcAnnotations.Initdata]; ok {
+		initdataTomlGzip, err := base64.StdEncoding.DecodeString(value)
+		if err != nil {
+			ociLog.Errorf("initdata uncompress failed! %v", err)
+			return err
+		}
+
+		gzipReader, err := gzip.NewReader(bytes.NewReader(initdataTomlGzip))
+		if err != nil {
+			return fmt.Errorf("create gzip reader failed: %v", err)
+		}
+		defer gzipReader.Close()
+
+		initdataToml, err := io.ReadAll(gzipReader)
+		if err != nil {
+			return fmt.Errorf("uncompressing initdata with gzip error: %v", err)
+		}
+
+		initdataStr := string(initdataToml)
+		var initdata kataTypes.Initdata
+		if _, err := toml.Decode(initdataStr, &initdata); err != nil {
+			return err
+		}
+
+		var initdataDigest []byte
+		switch initdata.Algorithm {
+		case "sha256":
+			h := sha256.New()
+			h.Write([]byte(initdataToml))
+			initdataDigest = h.Sum(nil)
```

Comment:
```suggestion
			initdataDigest = sha256.Sum256(initdataToml)
```
Nit: could be simplified.

Reply:
The return type of `sha256.Sum256()` is `[32]byte` not `[]byte` slice, thus we need
```go
sha256digest := sha256.Sum256(initdataToml)
initdataDigest = sha256digest[:]
```